storagenode/blobstore/filestore/dir.go

```diff
@@ -390,16 +392,15 @@ func (dir *Dir) TrashWithStorageFormat(ctx context.Context, ref blobstore.BlobRe
 		return err
 	}
 
-	// Change mtime to now. This allows us to check the mtime to know how long
-	// the file has been in the trash. If the file is restored this may make it
-	// take longer to be trashed again, but the simplicity is worth the
-	// trade-off.
+	// Change mtime to the logical time of removal. This allows us to check the
+	// mtime to know how long the file has been in the trash. If the file is
+	// restored this may make it take longer to be trashed again, but the
+	// simplicity is worth the trade-off.
 	//
 	// We change the mtime prior to moving the file so that if this call fails
 	// the file will not be in the trash with an unmodified mtime, which could
 	// result in its permanent deletion too soon.
-	now := dir.trashnow()
-	err = os.Chtimes(blobsVerPath, now, now)
+	err = os.Chtimes(blobsVerPath, withTimestamp, withTimestamp)
```

Comment:
This is the only time we use `trashnow`. I would prefer we use it if the timestamp is not set.

```suggestion
	if withTimestamp.IsZero() {
		withTimestamp = dir.trashnow()
	}
	err = os.Chtimes(blobsVerPath, withTimestamp, withTimestamp)
```

Reply:
@profclems, do you also want me to modify all call sites that set `withTimestamp` to `time.Now()` to use the zero value?

Reply:
Hey @liori, sorry for the late reply. I was out of office.

Actually, I think I would prefer we remove the `dir.trashnow`, since we don't use it. You can ignore my first suggestion.

Also a small nit, let's change `withTimestamp` to just `timestamp`. `withTimestamp` sounds more like a boolean value.

Reply:
Thank you, I've updated the PR with your recommendations.